components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -191,14 +235,16 @@ private void handleServerError(ActionInvocationResponse.Builder builder, HttpEnt
             if (errorResponse != null) {
                 builder.response(errorResponse);
             } else {
-                builder.errorLog("Failed to execute the action request. Received: " + statusCode);
+                builder.errorLog("Failed to execute the action request for status code " + statusCode
+                        + " due to no error response is available.");
                 builder.retry(true);
             }
         } catch (ActionInvocationException e) {
             //todo: add to diagnostic logs
             LOG.debug("JSON payload received for status code: " + statusCode +
                     " is not of the expected error response format. ", e);
-            builder.errorLog("Failed to execute the action request. Received: " + statusCode);
+            builder.errorLog("Failed to execute the action request for status code " + statusCode + ". "
+                    + e.getMessage());
```

Comment:
This should say an unexpected error response is recived with the status code

Reply:
Updated in